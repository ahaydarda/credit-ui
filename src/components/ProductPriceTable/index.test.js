
import React from "react";
import renderer from 'react-test-renderer';
import {ProductPriceTable} from "./index";
import {render, waitForElement, act} from '@testing-library/react'

const prices = [
    {
        "ticker": "ALPHA",
        "price": 3150.67,
        "assetClass": "Macro"
    },
    {
        "ticker": "BETA",
        "price": -350.67,
        "assetClass": "Credit"
    },
    {
        "ticker": "BETA",
        "price": 3791.37,
        "assetClass": "Equities"
    }
    ];
it('renders correctly', () => {
    const productPriceTable = renderer.create( <ProductPriceTable prices={prices} />)
        .toJSON();
    expect(productPriceTable).toMatchSnapshot();
});

it('shows negative prices with negative class', () => {
    const {container}= render( <ProductPriceTable prices={prices} />)
    const items = container.getElementsByClassName("negative-price")
    expect(items[0].textContent).toBe("-350.67");
});


it('shows positive prices with negative class', () => {
    const {container, getByText}= render( <ProductPriceTable prices={prices} />)
    const items = container.getElementsByClassName("positive-price");
    expect(items[0].textContent).toBe("3150.67");
    expect(items[1].textContent).toBe("3791.37");
});

