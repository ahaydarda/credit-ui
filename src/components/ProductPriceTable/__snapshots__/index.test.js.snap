// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="product-price-table"
>
  <div
    className="ctable"
  >
    <table
      style={
        Object {
          "gridTemplateColumns": "minmax(200px,4fr) minmax(150px,3fr) minmax(50px,3fr)",
        }
      }
    >
      <thead>
        <tr>
          <th>
            Asset Class
          </th>
          <th>
            Price
          </th>
          <th>
            Ticker
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="macro"
        >
          <td
            className="macro"
          >
            Macro
          </td>
          <td
            className="macro"
          >
            <span
              className="positive-price"
            >
              3150.67
            </span>
          </td>
          <td
            className="macro"
          >
            ALPHA
          </td>
        </tr>
        <tr
          className="credit"
        >
          <td
            className="credit"
          >
            Credit
          </td>
          <td
            className="credit"
          >
            <span
              className="negative-price"
            >
              -350.67
            </span>
          </td>
          <td
            className="credit"
          >
            BETA
          </td>
        </tr>
        <tr
          className="equities"
        >
          <td
            className="equities"
          >
            Equities
          </td>
          <td
            className="equities"
          >
            <span
              className="positive-price"
            >
              3791.37
            </span>
          </td>
          <td
            className="equities"
          >
            BETA
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
